src/FffGcodeWriter.h

```diff
@@ -480,17 +462,67 @@ class FffGcodeWriter : public SettingsMessenger, NoCopy
      * while newly generated perimeter gaps between consecutive insets of a concentric top/bottom pattern
      * are printed with the top bottom extruder.
      * 
+     * \param[in] storage where the slice data is stored.
      * \param gcode_layer The initial planning of the gcode of the layer.
      * \param mesh The mesh for which to add to the layer plan \p gcode_layer.
      * \param extruder_nr The extruder for which to print all features of the mesh which should be printed with this extruder
      * \param mesh_config the line config with which to print a print feature
      * \param skin_part The skin part for which to create gcode
-     * \param layer_nr The current layer number.
-     * \param skin_overlap The distance by which the skin overlaps with the wall insets and the distance by which the perimeter gaps overlap with adjacent print features.
-     * \param infill_angle The angle in the XY plane at which the infill is generated.
      * \return Whether this function added anything to the layer plan
      */
-    bool processSkinPart(const SliceDataStorage& storage, LayerPlan& gcode_layer, const SliceMeshStorage& mesh, const int extruder_nr, const PathConfigStorage::MeshPathConfigs& mesh_config, const SkinPart& skin_part, unsigned int layer_nr, int skin_overlap, int infill_angle) const;
+    bool processSkinPart(const SliceDataStorage& storage, LayerPlan& gcode_layer, const SliceMeshStorage& mesh, const int extruder_nr, const PathConfigStorage::MeshPathConfigs& mesh_config, const SkinPart& skin_part) const;
+
+    /*!
+     * Add the extra skin walls
+     * 
+     * \param[in] storage where the slice data is stored.
+     * \param gcode_layer The initial planning of the gcode of the layer.
+     * \param mesh The mesh for which to add to the layer plan \p gcode_layer.
+     * \param extruder_nr The extruder for which to print all features of the mesh which should be printed with this extruder
```

Comment:
No definitely not!
This is the extruder with which we are currently checking if we should print something.
It is not really the current extruder, because we might not yet have encountered any feature which caused us to actually switch to this extruder.

For each extruder we process each print feature and in each `processPrintFeatureX` function we check whether we should actually print something with the extruder.